core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1822,7 +1822,7 @@ class SparkContext(config: SparkConf) extends Logging {
       logInfo(s"Added file $path at $key with timestamp $timestamp")
       // Fetch the file locally so that closures which are run on the driver can still use the
       // SparkFiles API to access files.
-      Utils.fetchFile(uri.toString, root, conf, hadoopConfiguration, timestamp, useCache = false)
+      Utils.fetchFile(uri.toString, root, conf, hadoopConfiguration, timestamp, useCache = true)
```

Comment:
So are you saying that it doesn't fetch the file to `root` directory?

Reply:
> So are you saying that it doesn't fetch the file to `root` directory?

didn't copy as same filename

Reply:
Here is the log

Reply:
Can you point out which code we assign the temp name? `doFetchFile(url, targetDir, fileName, conf, hadoopConf)` should preserve the file name